finagle-core/src/main/scala/com/twitter/finagle/dispatch/ServerDispatcher.scala

```diff
@@ -1,11 +1,40 @@
 package com.twitter.finagle.dispatch
 
 import com.twitter.finagle.context.Contexts
-import com.twitter.finagle.transport.Transport
+import com.twitter.finagle.transport.{AnnotatedTransport, Transport}
 import com.twitter.finagle.{Service, NoStacktrace, CancelledRequestException}
 import com.twitter.util._
 import java.util.concurrent.atomic.AtomicReference
 
+/**
+  * A base Dispatcher type to allow for easy tracing upon `WireRecv` and
+  * `WireSend` by the 
+  * [[com.twitter.finagle.transport.Transport transporter]]. Clients should
+  * provided custom behavior via the handle and dispatch methods
+  * @tparam Req the type of Request
+  * @tparam Rep the type of Response
+  * @tparam In the type of input messages
+  * @tparam Out the type of output messages
+  * @param trans [[com.twitter.finagle.transport.AnnotatedTransport AnnotatedTransporter]] 
+  */
+abstract class ServerDispatcher[Req, Rep, In, Out](trans: AnnotatedTransport[In, Out])
```

Comment:
Yeah, I agree that it's unfortunate that the annotated-ness of the transport has to be exposed in the type system.  Purely abstractly, it would be nice if everything just took a Transport, and that could be either annotated or not.  I'm sure there are some details i'm handwaving over here.